core/src/RateModel.sol

```diff
@@ -3,27 +3,59 @@ pragma solidity 0.8.17;
 
```

Comment:
Idk why the git history on this file is so ugly. If you just open the old version and new version in separate windows it's much easier to see what's going on.